### PR TITLE
New VimL Cheat Sheet

### PR DESCRIPTION
**Pull request by phora**
https://duck.co/ia/view/viml_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@phora Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/viml.json [hunk 1]
```diff
@@ -0,0 +1,170 @@
+{
+    "id": "viml_cheat_sheet",
+    "name": "VimL",
+    "description": "Explanation of commonly used VimL methods",
+    "metadata": {
+		"sourceName": "Write a Vim script",
+		"sourceUrl": "http://vimdoc.sourceforge.net/htmldoc/usr_41.html"
+	},
+    "aliases": [
+        "viml", "vim scripting"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/viml.json]: @phora the `viml` alias isn't actually necessary :) -- the default trigger comes from the filename, so `viml cheat sheet` works by default.

File: share/goodie/cheat_sheets/json/viml.json [hunk 2]
```diff
@@ -0,0 +1,170 @@
+{
+    "id": "viml_cheat_sheet",
+    "name": "VimL",
+    "description": "Explanation of commonly used VimL methods",
+    "metadata": {
+		"sourceName": "Write a Vim script",
```
Comment (Justforahorror) [inline share/goodie/cheat_sheets/json/viml.json]: @phora  lets change this to "SourceForge"

Comment (jophab) [inline share/goodie/cheat_sheets/json/viml.json]: @phora Lets change this to "SourceForge"

File: share/goodie/cheat_sheets/json/viml.json [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    "template_type": "code",
+    "section_order": [
+        "Variable scope",
+        "Data types",
+        "Buffer interaction",
+        "String manipulation",
+        "I/O",
+        "List manipulation",
+        "Cursor manipulation"
+    ],
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/viml.json]: @phora I would capitalise the first letter of second word in section names.

File: share/goodie/cheat_sheets/json/viml.json [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            }
+        ],
+        "Cursor manipulation" : [
+            {
+                "key": "getpos('.')",
+                "val": "Get the position of mark . (the cursor)"
+            },
+            {
+                "key": "setpos('.', \\[0, 1, 1, 0\\])",
+                "val": "Set the position of mark . (the cursor) to line 1, column 1 in the current buffer"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/viml.json]: @phora  Is that full stop a typo?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=viml cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/viml_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=viml cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/viml_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/viml.json (hunk 1 above)
Comment (phora) [inline share/goodie/cheat_sheets/json/viml.json]: Gotcha! I'll be sure to update this in the next revision with more handy aliases.

File: share/goodie/cheat_sheets/json/viml.json (hunk 4 above)
Comment (phora) [inline share/goodie/cheat_sheets/json/viml.json]: No. The mark `. in vim represents the current cursor position.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=viml cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/viml_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/viml.json (hunk 4 above)
Comment (jophab) [inline share/goodie/cheat_sheets/json/viml.json]: @phora :+1:

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=viml cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/viml_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=viml cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/viml_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@phora this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by edgesince84** [comment]
@phora,  thanks again for creating the VimL Cheat Sheet! You’ve done a really good job here and we'd like you to help our community tackle the next big challenge - our mission to become the best search engine for programmers: https://duck.co/blog/post/297/help-for-programmers.

When you're ready to move on, please: 

- Create a new Instant Answer to increase our Programming IA coverage

OR

- Test live programming Instant Answers (or those in development) for bugs and suggest improvements 

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage).

Thanks again!

**Pull request merged**